packages/odyssey-react-mui/src/labs/DataComponents/TableSettings.tsx

```diff
@@ -113,15 +106,13 @@ const TableSettings = ({
         >
           <>
             {columns
-              .filter((column) => column.enableHiding !== false)
+              .filter((column) => Boolean(column.enableHiding))
               .map((column) => (
                 <MenuItem
                   key={column.accessorKey}
-                  onClick={changeColumnVisibility(column.id as string)}
+                  onClick={() => changeColumnVisibility(column.id!)}
```

Comment:
Yep, they're using useCallback

Reply:
They're not. Sadly, we need to memoize each child in a `useMemo` to fix that. It's so wonky, but that's the only way.

So the way I'd do it is memoize this part in `useMemo`:
```
columns
.filter((column) => Boolean(column.enableHiding))
```
And add `onClick` to the object.

Then when you `map` it here, it will have an already have a memoized `onClick`.